docs/cni-landscape.md

```diff
@@ -0,0 +1,50 @@
+# CNI Landscape
```

Comment:
I hope we don't get complaints if others deem this inaccurate or we don't keep it up-to-date...

Reply:
This was added by mistake and is still a work in progress. I will remove from this PR.